bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpUtil.java

```diff
@@ -180,13 +187,21 @@ public static String executeUrl(String httpMethod, String url, Properties httpHe
     private static ContentResponse executeUrlAndGetReponse(String httpMethod, String url, Properties httpHeaders,
             InputStream content, String contentType, int timeout, String proxyHost, Integer proxyPort, String proxyUser,
             String proxyPassword, String nonProxyHosts) throws IOException {
-        startHttpClient(CLIENT);
+
+        // Get shared http client from factory "on-demand"
+        if (HttpUtil.httpClient == null) {
+            // Bundle was not yet activated or has been deactivated - No better way to handle this case gracefully
```

Comment:
It isn't really about "bundle activation" - revise the comment.

Reply:
Agreed, I'll change comment to something like "referenced dependency was not (yet) injected"